### PR TITLE
[feature](array_cum_sum) Support array_cum_sum function

### LINKED ISSUE #17719: [Feature](array-function) Support array_cum_sum function
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

_No response_

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by brody715**
# Proposed changes

Issue Number: close #17719

## Problem summary

implement `array_cum_sum`

```shell
mysql> create table array_type_table(k1 INT, k2 Array<int>) duplicate key (k1) distributed by hash(k1) buckets 1 properties('replication_num' = '1');
mysql> insert into array_type_table values (0, []), (1, [NULL]), (2, [1, 2, 3, 4]), (3, [1, NULL, 3, NULL, 5]);
mysql> select k2, array_cum_sum(k2) from array_type_table;
+-----------------------+-----------------------+
| k2                    | array_cum_sum(`k2`)   |
+-----------------------+-----------------------+
| []                    | []                    |
| [NULL]                | [NULL]                |
| [1, 2, 3, 4]          | [1, 3, 6, 10]         |
| [1, NULL, 3, NULL, 5] | [1, NULL, 4, NULL, 9] |
+-----------------------+-----------------------+
```

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [x] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/array/function_array_cum_sum.cpp [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+        const auto& src_offsets = src_column_array->get_offsets();
+        const auto* src_nested_column = &src_column_array->get_data();
+        DCHECK(src_nested_column != nullptr);
+
+        // get src nested column
+        auto src_nested_type = assert_cast<const DataTypeArray&>(*src_arg.type).get_nested_type();
+
+        // get null map
+        const NullMapType* src_null_map = nullptr;
+        if (src_nested_column->is_nullable()) {
```
Comment (zhangstar333) [inline be/src/vec/functions/array/function_array_cum_sum.cpp:114]: could remove this check, now the nested column of array  must be nullable column in doris

Comment (brody715) [reply be/src/vec/functions/array/function_array_cum_sum.cpp]: Ok, I remove the check, and modify the source code

**Review by zhangstar333 — DISMISSED** [review]

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
@1430624563 this pr maybe have help for you

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.99 seconds
 stream load tsv:          424 seconds loaded 74807831229 Bytes, about 168 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          72 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230411032645_clickbench_pr_127179.html

**Comment by zhangstar333** [comment]
run p0

**Comment by brody715** [comment]
@zhangstar333 Hello, I've resolved the conflict to merge to the latest master right now. What do I need to do to continue the PR?

**Comment by eldenmoon** [comment]
LGTM

**Review by eldenmoon — DISMISSED** [review]

**Comment by eldenmoon** [comment]
ut seems failed, we could re push

**Comment by brody715** [comment]
> ut seems failed, we could re push

Sorry, I am new to contribute to this project. What does `re push` mean ? Does it mean I need to create a new commit and then push ?

**Comment by eldenmoon** [comment]
@brody715 Sorry, I didn't describe it clearly， you could rebase and push again

**Comment by brody715** [comment]
@eldenmoon thanks, re-push done

**Comment by eldenmoon** [comment]
run buildall

**Comment by brody715** [comment]
run p0

**Comment by brody715** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**